_candidates_ja/LucDuponcheel_1.md

```diff
@@ -0,0 +1,23 @@
+---
+name: Luc Duponcheel
+title: "記述的な方法による関数型プログラミング"
+length: 40
+audience: Advanced
+language: English
+twitter: LucDup
+github: LucDupAtGitHub
+icon: https://avatars0.githubusercontent.com/u/77954?s=400&u=7f67fc0da57a4ef97a65349c4f7cd39f40557812&v=4
+organization: ImagineJ
+tags:
+  - Functional Programming
+  - Best Practices
+suggestions:
+  - このトークを聞けば参加者は、高レベルな記述部分(description)と低レベル(でアクティブもしくはリアクティブ)な実行部分(interpretation)とを分離することによって実現されるプログラムの柔軟性を理解することが出来ます
+---
+チューリング章を受賞したJohn Backusは自身の講義のなかで、関数型プログラミング、すなわち関数、関数合成、ConstructionおよびConditionに基づく
+ポイントフリーな関数型プログラミングスタイルを提唱しました。
+関数型プログラミングはJohn Hughesが示したように、数学的な射影と深い関わりがあります。
+伝統的にポイントフリープログラミングは実現するには面倒なものだと考えられていましたが、
+Scalaを使えばDSLによってポイントフリープログラミングをほとんどネイティブなScalaプログラムと変わらない見た目に保つことが出来ます。
+このトークはライブラリのデザインにおける詳細および使用方法について話し、
+そしてimplicit関数がいかにしてボイラープレートなコードを削減するかを紹介します。
```

Comment:
The talk goes both into the details of the design of a library and it's usage. Among others it goes into details on how implicit functions can greatly reduce boilerplate code.